lib/launch_core/os/fs.c

```diff
@@ -27,24 +27,26 @@ int32_t l_fs_open(l_fs_file_t *p_file) { return fs_open(&p_file->file, p_file->f
 
 int32_t l_fs_close(l_fs_file_t *p_file) { return fs_close(&p_file->file); }
 
-size_t l_fs_write(l_fs_file_t *p_file, const uint8_t *const src) {
+size_t l_fs_write(l_fs_file_t *p_file, const uint8_t *const src, int32_t *err_flag) {
     if (!p_file->initialized) {
         LOG_ERR("Logger for file %s is not initialized", p_file->fname);
-        return -ENOTINIT;
+        *err_flag = -ENOTINIT;
+        return 0;
     }
 
     if ((p_file->wpos + p_file->width) > p_file->size) { // next write will go out-of-bounds
         fs_sync(&p_file->file);
         switch (p_file->mode) {
             case SLOG_ONCE: // error out
                 LOG_WRN("Logger reached end of file");
-                return -ENOSPC;
+                *err_flag = -ENOSPC;
+                return 0;
             case SLOG_CIRC: // wrap around
                 LOG_INF("Logger reset to start of file");
                 p_file->wpos = 0;
```

Comment:
Kind of outside of the scope of this PR but does the circular method have a way of storing the position of the "write head" to disk? If you timestamp the data its less of an issue but in an application where you don't have timestamps you don't know where your data starts/ends.

Reply:
Dont think so. Should be up to the user to have timestamps stored in their logs though

Reply:
Not really. The solution to your problem would be having a counter instead of a timestamp in that case